modules/apps/portal-search/portal-search-admin-web/src/main/java/com/liferay/portal/search/admin/web/internal/display/context/builder/IndexActionsDisplayContextBuilder.java

```diff
@@ -210,14 +211,24 @@ private Map<String, List<Object>> _getIndexersMap() {
 			List<Object> indexersData = new ArrayList<>();
 
 			for (Indexer<?> indexer : indexers) {
+				String className = indexer.getClassName();
+
+				if (className.contains(
+						"com.liferay.object.model.ObjectDefinition") &&
+					!_permissionChecker.isOmniadmin() &&
+					(indexer.getCompanyId() !=
+						CompanyThreadLocal.getCompanyId())) {
+
+					continue;
+				}
+
```

Comment:
looks pretty good, maybe instead of CompanyThreadLocal you could try _portal.getCompanyId(_httpServletRequest)

but i just had a thought when looking at this omniadmin check... we have the scope selector right, which allows an omni admin to select multiple companies... but the UI should probably be reactive to that right?... so maybe some show/hide logic should happen on front end? also, there is a _getInitialCompanyIds() method, so maybe that is something to look into as well as part of this first condition

Reply:
It was done by @oliv-yu the story as far as I know is completed we must just test to make sure it is 100%

Reply:
For the companyIds the logic with omniadmin is in the JS [here](https://github.com/liferay-search/liferay-portal/blob/a79c9432adc659e7e805854c7ab9e11e99cf9c51/modules/apps/portal-search/portal-search-admin-web/src/main/resources/META-INF/resources/js/IndexActions.js#L117-L121).

Hm, the `_getInitialCompanyIds` method was helpful to persist companyId selections after a page refresh from clicking on a reindex button, but I guess it isn't so important now after the component was refactored into react. It seemed harmless to keep in, except for now that it's confusing everyone 😅 

Asking @thektan to review my commits 😅